Installation issue: libsm

### Steps to reproduce the issue

```console
$ spack install libsm
```
fails to build with the error:
```
sm_genid.c:101:11: fatal error: uuid/uuid.h: No such file or directory
 # include <uuid/uuid.h>
           ^~~~~~~~~~~~~
compilation terminated.
```

### Platform and user environment

Please report your OS here:
```commandline
$ uname -a 
Linux emmet 4.18.0-147.3.1.el8_1.x86_64 #1 SMP Wed Nov 27 01:11:44 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ cat /etc/redhat-release
Red Hat Enterprise Linux release 8.1 (Ootpa)
``` 

### Additional information

Libuuid installs `include/uuid.h`, although on some platforms I've seen `/usr/include/uuid/uuid.h`.

```
$ spack spec libsm
Input spec
--------------------------------
libsm

Concretized
--------------------------------
libsm@1.2.2%gcc@8.3.1 arch=linux-rhel8-haswell
    ^libice@1.0.9%gcc@8.3.1 arch=linux-rhel8-haswell
        ^pkgconf@1.6.3%gcc@8.3.1 arch=linux-rhel8-haswell
        ^util-macros@1.19.1%gcc@8.3.1 arch=linux-rhel8-haswell
        ^xproto@7.0.31%gcc@8.3.1 arch=linux-rhel8-haswell
        ^xtrans@1.3.5%gcc@8.3.1 arch=linux-rhel8-haswell
    ^uuid@1.6.2%gcc@8.3.1 arch=linux-rhel8-haswell
```

--- a/var/spack/repos/builtin/packages/libsm/package.py
+++ b/var/spack/repos/builtin/packages/libsm/package.py
@@ -20,3 +20,4 @@ class Libsm(AutotoolsPackage, XorgPackage):
     depends_on('xtrans', type='build')
     depends_on('pkgconfig', type='build')
     depends_on('util-macros', type='build')
+    depends_on('libuuid')

